.github/workflows/csi-proxy.yml

```diff
@@ -0,0 +1,42 @@
+name: csi-proxy
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - hostprocess/csi-proxy/VERSION
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - hostprocess/csi-proxy/**
+    branchs:
+      - master
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: login to GitHub container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: set 'version' for ci
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          echo "version=ci" >> $GITHUB_ENV
+      - name: set 'version' for release
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          export version=$(cat ./hostprocess/csi-proxy/VERSION)
+          echo "version=$version" >> $GITHUB_ENV
+      - name: build image
+        run: |
+          cd hostprocess/csi-proxy
+          ./build.sh --push
```

Comment:
Should there be one more step here? one for build and one for push. Where push only runs `if: ${{ github.event_name != 'pull_request' }}`

Reply:
OOps, i think i meant to call `./build.sh` (without --push) for pull_requests and call `.build.sh --push) for non pull_requests.

Reply:
Updated